libibnetdisc/src/ibnetdisc.c

```diff
@@ -698,7 +709,12 @@ void add_to_portlid_hash(ibnd_port_t * port, GHashTable *htable)
 		/* We add the port for all lids
 		 * so it is easier to find any "random" lid specified */
 		for (lid = base_lid; lid <= (base_lid + lid_mask); lid++) {
-			g_hash_table_insert(htable, GINT_TO_POINTER(lid), port);
+			struct lid2guid_item *item;
+
+			item = malloc(sizeof(*item));
+			if (item)
+				cl_qmap_insert(&f_int->lid2guid, lid,
```

Comment:
I believe you need to set item->port = port before the insert.

I'm also not a huge fan of having to allocate here...  But there is nothing we can do about it other than having an internal ibnd_port_t object which has the cl_map_item_t in it.  Perhaps we should just make a comment that we don't think the alloc will be enough of a problem to justify the complexity of an other 'internal' data structure.

Reply:
oops, yes that assignment is needed